app/controllers/subscriptions_management_controller.rb

```diff
@@ -0,0 +1,149 @@
+class SubscriptionsManagementController < ApplicationController
+  before_action :set_cache_control_header
+  before_action :check_if_access_is_enabled
+  before_action :get_subscriptions
+
+  MISSING_EMAIL_ERROR = "Please enter your email address.".freeze
+  INVALID_EMAIL_ERROR = "This doesn’t look like a valid email address – check you’ve entered it correctly.".freeze
+
+  def index; end
+
+  def update_frequency
+    @back_url = list_subscriptions_path
+    @subscription_id = subscription_id
+    @current_frequency = @subscriptions[@subscription_id]['frequency']
+  end
+
+  def change_frequency
+    # TODO: Enable this
+    # email_alert_api.change_subscription(
+    #   id: subscription_id,
+    #   frequency: new_frequency
+    # )
+    session[:subscriptions][subscription_id]['frequency'] = new_frequency
+
+    #clear_subscriptions!
+
+    redirect_to list_subscriptions_path
+  end
+
+  def update_address
+    @back_url = list_subscriptions_path
+    @address = @subscriber['address']
+  end
+
+  def change_address
+    @back_url = list_subscriptions_path
+    @address = @subscriber['address']
+
+    unless subscriber_params[:new_address].present?
+      flash.now[:error] = MISSING_EMAIL_ERROR
+      render :update_address
+      return
+    end
+
+    begin
+      # TODO: Enable this
+      # email_alert_api.change_subscriber(
+      #   id: subscriber_id,
+      #   new_address: new_address
+      # )
+    rescue GdsApi::HTTPUnprocessableEntity
+      @new_address = new_address
+      flash.now[:error] = INVALID_EMAIL_ERROR
+      render :update_address
+      return
+    end
+
+    session[:subscriber]['address'] = new_address
+
+    #clear_subscriptions!
+
+    redirect_to list_subscriptions_path
+  end
+
+private
+
+  def check_if_access_is_enabled
+    render plain: '404 Not Found', status: 404 unless ENV['SUBSCRIPTION_MANAGEMENT_ENABLED']
+  end
+
+  def set_cache_control_header
+    headers["Cache-Control"] = "private"
+  end
+
+  def get_subscriptions
+    # TODO: Hardcoded for now
+    #email_alert_api.get_subscriptions(id: subscriber_id)
+    if session[:subscriber].nil? || session[:subscriptions].nil?
+      @_subscription_details = {
+        "subscriber": {
+          "address": "hong@jabra.com"
```

Comment:
😆

Reply:
Haha yeah, unfortunately removed now for the real thing